src/components/common/Header/index.tsx

```diff
@@ -3,17 +3,22 @@ import styled from "styled-components";
 import { theme, flex } from "@/styles";
 import { Logo, Setting } from "@/assets/icons";
 import Navigation from "./Navigation";
+import SettingModal from "./@setting/layouts/setting";
+import { useSettingModal } from "./@setting/hooks";
 
 const Header = () => {
+  const { isSettingModalOpen, openSettingModal, closeSettingModal } =
+    useSettingModal();
+
   return (
     <Layout>
       <Logo width={42} />
       <Navigation />
-      <Setting />
+      <Setting onClick={openSettingModal} />
+      {isSettingModalOpen && <SettingModal onClose={closeSettingModal} />}
     </Layout>
```

Comment:
```suggestion
import { useModal } from "@/@modal/hooks";

const Header = () => {
    const { openModal } = useModal;
    
    const handleOpenSettingModalClick = () => {
         openModal({ component: <SettingModal/> });
    }
    
    return ... <Setting onClick={handleOpenSettingModalClick} />
}
```

위에서 말씀드린 제가 미리 만들어둔 전역 모달 훅을 사용하면 이렇게 훨씬 편하게 작성할 수 있답니다...!!!
그래도 모달 만든다고 수고하셨습니다!! 되게 값진 경험 되었을 것 같아요 굳굳